### DISCUSSION TITLE
Works only with ili9341 (not st7796s not ili9488)

### DISCUSSION THREAD

**Discussion by manuferro** [Bodmer/TFT_eSPI#1146] [Compatible displays and setup files]
Hi,
  I've some problem with esp32 and st7796s 
  I'm able to make ili9341 works. 
I'm not able to make the same with st7796s and ili9488. [link to display](https://it.aliexpress.com/item/4000817659016.html)
always blank screen (just the backlight is on!)
The only wiring difference between ili9341 & st7796s/ili9488 is related to IM0/1/2. With st7796s/ili9488 all the 3 are connected to VCC (while with ili9341 the IM0 is connected to GND)

I've done tooons of test. (also with ESPIDF)
The problems seems the SPI communication  but verifying with a digital analyzer everything seems ok.
I'm even not able to read the screen ID.

Do somebody has any idea about what I can check?

![Evernote_ijUj7QXhxT](https://user-images.githubusercontent.com/6489807/115138880-161f7700-a02f-11eb-8bdc-2d4923b951c6.jpg)

**Comment by manuferro** [comment]
From Read_User_setup: 
(same wiring is working with Ili9341 obviously with a different driver) 
18:00:33.661 -> [code]
18:00:33.661 -> TFT_eSPI ver = 2.3.60
18:00:33.661 -> Processor    = ESP32
18:00:33.695 -> Frequency    = 80MHz
18:00:33.695 -> Transactions = Yes
18:00:33.695 -> Interface    = SPI
18:00:33.695 -> Display driver = 7796
18:00:33.695 -> Display width  = 320
18:00:33.695 -> Display height = 480
18:00:33.695 -> 
18:00:33.695 -> MOSI    = GPIO 23
18:00:33.695 -> MISO    = GPIO 19
18:00:33.695 -> SCK     = GPIO 18
18:00:33.695 -> TFT_CS   = GPIO 5
18:00:33.695 -> TFT_DC   = GPIO 2
18:00:33.695 -> TFT_RST  = GPIO 27
18:00:33.695 -> 
18:00:33.695 -> TFT_BL           = GPIO 4
18:00:33.695 -> 
18:00:33.695 -> Font GLCD   loaded
18:00:33.695 -> Font 2      loaded
18:00:33.695 -> Font 4      loaded
18:00:33.695 -> Font 6      loaded
18:00:33.695 -> Font 7      loaded
18:00:33.728 -> Font 8      loaded
18:00:33.728 -> Smooth font enabled
18:00:33.728 -> 
18:00:33.728 -> Display SPI frequency = 20.00
18:00:33.728 -> [/code]

**Comment by Bodmer** [comment]
I have it working on both those types of display. I suspect it is an interface configuration problem at hardware level. The setup file output looks fine.
